R/utils.R

```diff
@@ -63,3 +67,21 @@ color_yiq <- function(r, g, b) {
 color_yiq_islight <- function(r, g, b, threshold = 150) {
   color_yiq(r, g, b) >= threshold
 }
+
+
+# Wrapper around base::system.file. In base::system.file, the package directory
+# lookup is a bit slow. This caches the package directory, so it is much faster.
+system_file <- local({
+  package_dir_cache <- character()
+
+  function(..., package = "base") {
+    if (package %in% names(package_dir_cache)) {
+      package_dir <- package_dir_cache[[package]]
+    } else {
+      package_dir <- system.file(package = package)
+      package_dir_cache[[package]] <<- package_dir
+    }
+
```

Comment:
This could be problematic if we ever want to use other named args `system.file()`...perhaps we can just add `ellipsis::check_dots_unnamed()`?

Reply:
Good thinking. I'll implement something similar so we don't have to bring in the ellipsis package.